db/db_impl/db_impl.cc

```diff
@@ -4328,6 +4328,20 @@ void DBImpl::ReleaseSnapshot(const Snapshot* s) {
       }
       bottommost_files_mark_threshold_ = new_bottommost_files_mark_threshold;
     }
+
+    for (auto* cfd : *versions_->GetColumnFamilySet()) {
```

Comment:
This part probably needs some optimization. Going through all column families regardless is not nice. I will work on some global threshold type of tracking for this.